src/lib/ngsiNotify/Notifier.cpp

```diff
@@ -1,569 +1,570 @@
+    // soon (it has been scheduled for version 0.19.0, see https://github.com/telefonicaid/fiware-orion/issues/714)
+    // we introduce the atLeastOneNotDefault hack. Once #714 gets implemented,
+    // this FIXME will be removed (and all the test harness adjusted, if needed)
+    //
+    if (!atLeastOneNotDefault)
+    {
+      spathList = "";
+    }
+
+    ci.outMimeType = JSON;
+
+    std::string _payloadString;
+    if (renderFormat == NGSI_V1_LEGACY)
+    {
+      bool asJsonObject = (ci.uriParam[URI_PARAM_ATTRIBUTE_FORMAT] == "object" && ci.outMimeType == JSON);
+      _payloadString = ncrP->render(ci.apiVersion, asJsonObject);
+    }
+    else
+    {
+      _payloadString = ncrP->toJson(renderFormat, attrsOrder, metadataFilter, blackList);
+    }
+
+    std::string payloadString;
+#ifdef PARANOID_JSON_INDENT
+    if (paranoidV1Indent && (renderFormat == NGSI_V1_LEGACY))
+    {
+      // First stage: conventional pretty printer
+      rapidjson::Document doc;
+      doc.Parse(_payloadString.c_str());
+
+      rapidjson::StringBuffer s;
+      rapidjson::PrettyWriter<rapidjson::StringBuffer> writer(s);
+      writer.SetIndent(' ', 2);
+      doc.Accept(writer);
+
+      std::string prettyPrinted = s.GetString();
+
+      // Second stage: "key": "value" -> "key" : value (legacy reasons... this was the
+      // way we implement JSON rendering at the very beggining)
+      char* prettyPrinted2 = jsonFix(prettyPrinted.c_str());
+      payloadString = std::string(prettyPrinted2);
+      free(prettyPrinted2);
+    }
+    else
+    {
+      payloadString = _payloadString;
+    }
+#else
+    payloadString = _payloadString;
+#endif
+
+    /* Parse URL */
+    std::string  host;
+    int          port;
+    std::string  uriPath;
+    std::string  protocol;
+
+    if (!parseUrl(httpInfo.url, host, port, uriPath, protocol))
+    {
+      LM_E(("Runtime Error (not sending NotifyContextRequest: malformed URL: '%s')", httpInfo.url.c_str()));
+      return paramsV;  //empty vector
+    }
+
+    /* Set Content-Type */
+    std::string content_type = "application/json";
+
+
+    SenderThreadParams*  params = new SenderThreadParams();
+
+    params->ip               = host;
+    params->port             = port;
+    params->protocol         = protocol;
+    params->verb             = verbName(verb);
+    params->tenant           = tenant;
+    params->servicePath      = spathList;
+    params->xauthToken       = xauthToken;
+    params->resource         = uriPath;
+    params->content_type     = content_type;
+    params->content          = payloadString;
+    params->mimeType         = JSON;
+    params->renderFormat     = renderFormatToString(renderFormat);
+    params->fiwareCorrelator = fiwareCorrelator;
+    params->subscriptionId   = ncrP->subscriptionId.get();
+    params->registration     = false;
+
+    paramsV->push_back(params);
+    return paramsV;
+}

```

Comment:
Why do we see the entire file as modified?
Can only suspect windows line feed/carriage return ...
It's not possible to view the real change like this, please try and redo the chamges to this file so that the diff shows only the affected lines.

Reply:
Fixed in 273e069